Updates to install/configuration (#37)
AdobeDocs/commerce-extensibility: src/pages/events/configure-commerce.md

---
title: Install and configure Adobe I/O Events
description: Learn how to install the Commerce modules needed for Adobe I/O Events and configure both Commerce and your existing Adobe App Builder project.
---

# Install and configure Adobe I/O Events

After you've created an [App Builder project](./project-setup.md), you must install the Commerce modules that enable integrations with Adobe I/O Events.

## Install Adobe I/O modules on Commerce

Make the following modifications to your `composer.json` file:

> *  Add a `repositories` section beneath the `config` section. If the `repositories` section already exists, add the lines to the bottom of the section.

Reviewer: ```suggestion
*  Add a `repositories` section beneath the `config` section. If the `repositories` section already exists, add the following lines to the bottom of the section.
```

   ```json
   "repositories": {

>     "module-adobe-io-events": {

Reviewer 2: Please add additional spaces before the repositories name, because `"repositories"` and `"module-adobe-io-events"` are almost on the same level
```json
"repositories": {
  "module-adobe-io-events": {
    "type": "git",
    "url": "git@github.com:magento/module-adobe-io-events.git"
  },
  "event-plugin-generator": {
    "type": "git",
    "url": "git@github.com:magento/event-plugin-generator.git"
  },
  "module-commerce-events-client": {
    "type": "git",
    "url": "git@github.com:magento/module-commerce-events-client.git"
  }
},
```

        "type": "git",
        "url": "git@github.com:magento/module-adobe-io-events.git"
    },
    "event-plugin-generator": {
        "type": "git",
        "url": "git@github.com:magento/event-plugin-generator.git"
    },
    "module-commerce-events-client": {
        "type": "git",
        "url": "git@github.com:magento/module-commerce-events-client.git"
    }
   },
   ```

> *  Add the following lines to the bottom of the `require` section.

Reviewer: ```suggestion
*  Add the following lines to the bottom of the `require` section:
```

   ```json
   "magento/module-adobe-io-events": "dev-master as 0.99",
   "magento/module-commerce-events-client": "dev-main as 0.99",
   "magento/event-plugin-generator": "dev-main as 0.99"
   ```

*  Find the `prefer-stable` line and modify the minimum stability:

   ```json
   "prefer-stable": true,
   "minimum-stability": "dev"
   ```

Save your changes. The remaining installation steps vary, depending on your environment.

### Cloud installation

1. Update your `.magento.app.yaml` file to include the following build hook:

```yaml
hooks:
  build: |
   set -e
   composer install
   php ./bin/magento events:generate:module
   php ./bin/magento module:enable Magento_AdobeCommerceEvents
   php ./bin/magento module:enable Magento_AdobeCommerceEventsClient
   php ./bin/magento module:enable Magento_AdobeCommerceEventsGenerator
   php ./vendor/bin/ece-tools run scenario/build/generate.xml
   php ./vendor/bin/ece-tools run scenario/build/transfer.xml
```

> This hook generates and enables the `AdobeCommerceEvents` module, which allows you to register custom events.

Reviewer: Should this be indented as part of step #1?

<!--Note to reviewer: I deleted references of bin/magento events:metadata:populate because I'm going to assume that a custom module with specific pre-defined events isn't available at initial installation/configuration. I will create a separate "Module development/integration" topic that addresses this command. -->

### Local and on-premises installation

1. Update the project dependencies.

  ```bash
  composer update
  ```

> 1. Run the following command to generate the `AdobeCommerceEvents` module. This module helps register  events. At this point, the module is empty.

Reviewer: ```suggestion
1. Run the following command to generate the `AdobeCommerceEvents` module. This module helps register events. Initially, the module is empty.
```

   ```bash
   bin/magento events:generate:module
   ```

> 1. Enable the newly-installed modules:

Reviewer: ```suggestion
1. Enable the newly installed modules:
```

   ```bash
   bin/magento module:enable --all
   ```

1. Upgrade your instance:

   ```bash
   bin/magento setup:upgrade
   ```

1. Compile your instance to generate new classes:

   ```bash
   bin/magento setup:di:compile
   ```

## Begin configuring events on Commerce

> You must configure Commerce to communicate with your project. You will need two files that you downloaded from the the Adobe Developer Console.

Reviewer: ```suggestion
You must configure Commerce to communicate with your project. You will the need two files that you downloaded from the the Adobe Developer Console.
```

Should we name the files here? or maybe point to the sections where downloading each file is discussed ([Set up a project](https://developer-stage.adobe.com/commerce/extensibility/events/project-setup/#set-up-a-project) (last step) and [Download the workspace configuration file](https://developer-stage.adobe.com/commerce/extensibility/events/project-setup/#download-the-workspace-configuration-file)

1. In the Commerce Admin, navigate to **Stores** > Settings > **Configuration** > **Adobe Services** > **Adobe I/O Events** > **General configuration**.

1. Copy and paste the contents of the `private.key` file into the **Service Account Private Key** field. Use the following command to copy the contents.

   ```bash
   cat config/private.key | pbcopy
   ```

1. Copy the contents of the `<workspace-name>.json` file into the **Adobe I/O Workspace Configuration** field.

1. Enter a unique identifier in the **Adobe Commerce Instance ID** field. This value can be any unique string.

1. Click **Save Config**, but do not leave the page. The next section creates an event provider, which is necessary to complete the configuration.

> ## Create an event provider and finish Commerce configuration

Reviewer: ```suggestion
## Create an event provider and complete the Commerce configuration
```

You cannot create an event provider until you have configured and saved a private key, workspace file, and instance ID values.

1. Create a `<Commerce-root-directory>/app/etc/event-types.json` file and add the following:

   ```json
   {
    "provider": {
        "label": "My Adobe Commerce Events",
        "description": "Provides out-of-process extensibility for Adobe Commerce"
        }
    }
    ```

1. Run the following command to create an event provider:

   ```bash
   bin/magento events:create-event-provider
   ```

   The command displays a message similar to the the following:

   ```terminal
   No event provider found, a new event provider will be created
   A new event provider has been created with ID 63a1f8fe-e911-45a4-9d3f
   ```

> 

Reviewer 2: As PR https://github.com/magento-commerce/module-adobe-io-events/pull/3 was merged now we can simplify and remove step 1 and leave only the next command: 
```
bin/magento events:create-event-provider --label='Provider label' --description='Provider description'
```

1. Copy the ID returned in the command output into the **Adobe I/O Event Provider ID** field in the Admin.

> 1. Enter the provided URL as the value of the **Endpoint** fields.

Reviewer: ```suggestion
1. Enter the provided URL as the value of the **Endpoint** field.
```
